lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -1379,7 +1388,7 @@ LowererMDArch::LowerEntryInstr(IR::EntryInstr * entryInstr)
     //
     uint32 argSlotsForFunctionsCalled = this->m_func->m_argSlotsForFunctionsCalled;
     // Stack is always reserved for at least 4 parameters.
-    if (argSlotsForFunctionsCalled < 4)
+    if (argSlotsForFunctionsCalled < 4 && IsArgSaveRequired(this->m_func))
```

Comment:
The initialization value for argSlotsForFunctionCalled is read from the Func constructor which sets it to 0.